resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverBuilder.java

```diff
@@ -291,14 +303,17 @@ public DnsNameResolverBuilder hostsFileEntriesResolver(HostsFileEntriesResolver
      * @return a {@link DnsNameResolver}
      */
     public DnsNameResolver build() {
+
+        if (resolveCache != null && (minTtl != 0 || maxTtl != Integer.MAX_VALUE || negativeTtl != 0)) {
+            throw new IllegalArgumentException("resolveCache and TTLs are mutually exclusive");
```

Comment:
This won't work if a user called `minTtl(0)`, `maxTtl(Integer.MAX_VALUE)`, etc. `minTtl/maxTtl,negativeTtl` needs to be `Integer` rather that `int`.

Also, please throw `IllegalStateException`.